system/Superglobals.php

```diff
@@ -16,48 +16,381 @@
 /**
  * Superglobals manipulation.
  *
+ * Provides a clean API for accessing and manipulating PHP superglobals
+ * with support for testing and backward compatibility.
+ *
+ * Note on return types:
+ * - $_SERVER can contain int (argc, REQUEST_TIME) or float (REQUEST_TIME_FLOAT)
+ * - $_SERVER['argv'] is an array
+ * - $_GET, $_POST, $_REQUEST can contain nested arrays from query params like ?foo[bar]=value
+ * - $_COOKIE typically contains strings but can have arrays with cookie[key] notation
+ *
  * @internal
  * @see \CodeIgniter\SuperglobalsTest
  */
 final class Superglobals
 {
+    /**
+     * @var array<string, array|float|int|string>
+     */
     private array $server;
+
+    /**
+     * @var array<string, array|string>
+     */
     private array $get;
 
-    public function __construct(?array $server = null, ?array $get = null)
-    {
-        $this->server = $server ?? $_SERVER;
-        $this->get    = $get ?? $_GET;
+    /**
+     * @var array<string, array|string>
+     */
+    private array $post;
+
+    /**
+     * @var array<string, array|string>
+     */
+    private array $cookie;
+
+    /**
+     * @var array<string, array|string>
+     */
+    private array $request;
+
+    /**
+     * @var array<string, array<string, mixed>>
+     */
+    private array $files;
+
+    /**
+     * @param array<string, array|float|int|string>|null $server
+     * @param array<string, array|string>|null           $get
+     * @param array<string, array|string>|null           $post
+     * @param array<string, array|string>|null           $cookie
+     * @param array<string, array<string, mixed>>|null   $files
+     * @param array<string, array|string>|null           $request
+     */
+    public function __construct(
+        ?array $server = null,
+        ?array $get = null,
+        ?array $post = null,
+        ?array $cookie = null,
+        ?array $files = null,
+        ?array $request = null,
+    ) {
+        $this->server  = $server ?? $_SERVER;
+        $this->get     = $get ?? $_GET;
+        $this->post    = $post ?? $_POST;
+        $this->cookie  = $cookie ?? $_COOKIE;
+        $this->files   = $files ?? $_FILES;
+        $this->request = $request ?? $_REQUEST;
     }
 
-    public function server(string $key): ?string
+    /**
+     * Get a value from $_SERVER
+     *
+     * @return array<array-key, mixed>|float|int|string|null
+     */
+    public function server(string $key): array|float|int|string|null
```

Comment:
For the getters, how about the idea of introducing a default parameter, liked `mixed $default = null` so that on invocation we can define already the default value in case missing instead of doing `?? <default>`?